websites/F/Free Web Novel/presence.ts

```diff
@@ -0,0 +1,192 @@
+const presence = new Presence({
+		clientId: "965294297048023050"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	let [showCover, showButtons, showBook, showLogo] = await Promise.all([
+		presence.getSetting<boolean>("showCover"),
+		presence.getSetting<boolean>("showButtons"),
+		presence.getSetting<boolean>("showBook"),
+		presence.getSetting<boolean>("logoOnly")
+	]);
+	const [privacy, showTimestamp, showReading] = await Promise.all([
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("showTimestamp"),
+			presence.getSetting<boolean>("showReading")
+		]),
+		presenceData: PresenceData = {
+			largeImageKey: "nocover"
+		},
+		{ pathname } = window.location,
+		pathnames = [
+			"/",
+			"/history/",
+			"/latest-release-novel/",
+			"/latest-novel/",
+			"/completed-novel/",
+			"/most-popular-novel/",
+			"/search/",
+			"/contact.html",
+			"/sitemap.xml",
+			"/privacy-terms-of-use.html",
+			"/genre/",
+			"/author/"
+		];
+	if (!showReading) {
+		showBook = false;
+		showCover = false;
+		showLogo = true;
+		showButtons = false;
+	} else {
+		presenceData.smallImageKey = "closed";
+		presenceData.smallImageText = "Not Reading";
+	}
+	if (showTimestamp) presenceData.startTimestamp = browsingTimestamp;
+	if (!showBook) showCover = false;
+	switch (true) {
+		case privacy: {
+			presenceData.largeImageKey = "incognito";
+
+			if (
+				(pathnames.includes(pathname) ||
+					document.querySelector<HTMLDivElement>('[class="m-imgtxt"]')) &&
+				showReading
+			)
+				presenceData.details = "Browsing...";
+			else if (
+				document.querySelector<HTMLAnchorElement>(
+					'[title="Read Next chapter"]'
+				) &&
+				showReading
+			) {
+				presenceData.details = "Reading...";
+				presenceData.smallImageKey = "open";
+			}
+			break;
```

Comment:
this should be one of the last cases because things after this line might be true and might overwrite what this is doing. This doesn't work exactly like a chain of else ifs